src/lib/service/learnerRecordAPI/event/service.ts

```diff
@@ -0,0 +1,44 @@
+import {User} from 'src/lib/model'
+import {CourseRecord} from '../courseRecord/models/courseRecord'
+import {RecordState} from '../models/record'
+import {BookingStatus, ModuleRecord} from '../moduleRecord/models/moduleRecord'
+import {getEventsByUids} from './client'
+import {Event} from './models/event'
+
+export async function getCancelledEventUidsFromCourseRecord(courseRecords: CourseRecord | CourseRecord[], user: User) {
+	let records: CourseRecord[]
+	if (Array.isArray(courseRecords)) {
+		records = courseRecords
+	} else {
+		records = [courseRecords]
+	}
+
+	const eventIds = getEventUidsFromCourseRecords(records)
+	const learnerRecordEvents: Event[] = await getEventsByUids(eventIds, user)
```

Comment:
We shouldn't be accessing the API if `eventIds` is empty. This is causing an exception when I load the homepage with nothing in my learning plan